v3/tasks/20250321-v3-vault-factory-v2/readme.md

```diff
@@ -0,0 +1,10 @@
+# 2025-03-10 - V3 VaultFactory V2
+
+Redeployment of the VaultFactory for V3.
+The current VaultFactory deploys the ProtocolFeeController and passes it into the Vault. This introduces an unwanted coupling, since with this design we would need to redeploy the VaultFactory every time we migrated to a new version of the fee controller. This version removes the deployment and accepts the fee controller address in the `create` function, passing it through to the Vault. (This will also decrease the size of the VaultFactory contract.)
```

Comment:
```suggestion
The current VaultFactory deploys the ProtocolFeeController and passes it into the Vault. This introduces an unwanted coupling, since with this design we would need to redeploy the VaultFactory every time we migrated to a new version of the fee controller. This version removes the deployment and accepts the fee controller address in the `create` function, passing it through to the Vault. (This will also decrease the size of the VaultFactory contract and the gas limit to deploy the Vault through the factory.)
```